src/Cake.Issues/IIssueComparer.cs

```diff
@@ -0,0 +1,135 @@
+﻿namespace Cake.Issues
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Comparer to compare if two issues are identical.
+    /// </summary>
+    public class IIssueComparer : IEqualityComparer<IIssue>
+    {
+        private readonly bool compareOnlyPersistentProperties;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IIssueComparer"/> class.
+        /// Two issues are seen as identical if all properties have identical values.
+        /// </summary>
+        public IIssueComparer()
+            : this(false)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IIssueComparer"/> class.
+        /// </summary>
+        /// <param name="compareOnlyPersistentProperties">Flag indicating whether properties which
+        /// are affected by changes in files should be considered while comparing issues.
+        /// If set to <c>true</c>, the comparer can be used to compare issues from different
+        /// build runs, where files might have been changed or renamed.</param>
+        /// <remarks>
+        /// If <paramref name="compareOnlyPersistentProperties"/> is set to <c>true</c> the following
+        /// properties will be ignored while comparing the issue:
+        /// <list type="bullet">
+        /// <item>
+        /// <description><see cref="IIssue.ProjectFileRelativePath"/></description>
+        /// </item>
+        /// <item>
+        /// <description><see cref="IIssue.AffectedFileRelativePath"/></description>
+        /// </item>
+        /// <item>
+        /// <description><see cref="IIssue.Line"/></description>
+        /// </item>
+        /// </list>
+        /// </remarks>
+        public IIssueComparer(bool compareOnlyPersistentProperties)
+        {
+            this.compareOnlyPersistentProperties = compareOnlyPersistentProperties;
+        }
+
+        /// <inheritdoc/>
+        public bool Equals(IIssue x, IIssue y)
+        {
+            if (ReferenceEquals(x, y))
```

Comment:
`object.ReferenceEquals(x, y)` would be more robust. But for this class this should be ok. Your decision if you want to change this.